build-tools/automation/build.groovy

```diff
@@ -65,18 +83,48 @@ timestamps {
 
             def buildType = isPr ? 'PR' : 'CI'
 
+            echo "Git repo: ${env.GitRepo}"     // Defined as an environment variable in the jenkins build definition
             echo "Job: ${env.JOB_BASE_NAME}"
             echo "Branch: ${branch}"
             echo "Commit: ${commit}"
             echo "Build type: ${buildType}"
+
             if (isPr) {
                 echo "PR id: ${env.ghprbPullId}"
                 echo "PR link: ${env.ghprbPullLink}"
+
+                // Clear out the PR title and description. This is the equivalent of $JENKINS_HOME/global-pre-script/remove-problematic-ghprb-parameters.groovy used by freestyle builds
+                echo "Clearing the PR title and description environment variables to avoid any special characters contained within from tripping up the build"
+                env.ghprbPullTitle = ''
+                env.ghprbPullLongDescription = ''
+
+                if (hasPrLabel(env.GitRepo, env.ghprbPullId, 'full-mono-integration-build')) {
+                    buildTarget = 'jenkins'
+                } else {
+                    buildTarget = 'all'
+                }
             }
+
+            echo "${buildType} buildTarget: ${buildTarget}"
```

Comment:
I'm not sure we need this message per-se, as it'll be easily deducible from the later `make ${buildType}` invocation.

Reply:
I prefer to keep it. The information is logged in the init stage alongside other settings pertinent to the build